chapter_6/ex01_doxdocgen/include/chapter6/ex01/calculator.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * ______________________________________________________
+ * @file calculator.hpp
+ *
+ * Calculator interface declaration
+ *
+ * SPDX-License-Identifier:	MIT
+ * ______________________________________________________
+ */
+
+#pragma once
+
+#include <chapter6/ex01/calculator_interface.hpp>
+
+namespace chapter6 {
+namespace ex01 {
+/**
+ * @brief The basic 'calculator' class
+ *
+ * Implements the @ref calculator_interface to provide the four
+ * basic arithmetic operations.
+ */
+class calculator : private calculator_interface {
+public:
+  /**
+   * Calculate the sum of two numbers, @p augend lhs and @p addend
+   *
+   * @param [in] augend The number to which @p addend is added
+   * @param [in] addend The number which is added to @p augend
+   *
+   * @return double Sum of two numbers, @p lhs and @p rhs
+   */
+  virtual double sum(double augend, double addend) override;
+
+  /**
+   * Calculate the difference of @p rhs from @p lhs
+   *
+   * @param [in] minuend    The number to which @p subtrahend is subtracted
+   * @param [in] subtrahend The number which is to be subtracted from @p minuend
+   *
+   * @return double Difference of two numbers, @p minuend and @p subtrahend
+   */
+  virtual double sub(double minuend, double subtrahend) override;
```

Comment:
`override` implies `virtual` so no need to state it twice.

Reply:
I like being explicit.

Reply:
It's not wrong, it's just redundant ;)